packages/profiling-node/src/integration.ts

```diff
@@ -305,6 +346,30 @@ class ContinuousProfiler {
     };
   }
 
+  /**
+   * Assigns thread_id and thread name context to a profiled event.
+   */
+  private _assignThreadIdContext(event: Event): any {
+    if (!event?.['contexts']?.['profile']) {
+      return;
+    }
+
+    if (!event.contexts) {
+      return;
+    }
+
+    // @ts-expect-error the trace fallback value is wrong, though it should never happen
+    // and in case it does, we dont want to override whatever was passed initially.
+    event.contexts['trace'] = {
+      ...(event.contexts?.['trace'] ?? {}),
+      data: {
+        ...(event.contexts?.['trace']?.['data'] ?? {}),
+        ['thread.id']: PROFILER_THREAD_ID_STRING,
+        ['thread.name']: PROFILER_THREAD_NAME,
```

Comment:
do we also want to attach this to every span?

should we also only set `thread` attributes if trace context is defined?

Reply:
Right now not yet, but with span streaming we would want to do that yes